# Description: Remove rubygems from various places in Chef were not needed.
# Also set up chef-server programs to load slices correctly w/o rubygems.
# Upstream does not plan to remove rubygems, so this patch needs to be
# updated with new files that require rubygems.
Index: chef-0.8.14+dfsg.orig/chef-server-api/config/init.rb
===================================================================
--- chef-0.8.14+dfsg.orig.orig/chef-server-api/config/init.rb	2010-05-07 15:06:33.000000000 -0600
+++ chef-0.8.14+dfsg.orig/chef-server-api/config/init.rb	2010-05-07 15:07:01.000000000 -0600
@@ -29,8 +29,6 @@
 dependency "merb-helpers", merb_gems_version
 dependency "chef", :immediate=>true unless defined?(Chef)
 
-require 'rubygems'
-
 Merb::Config.use do |c|
   c[:session_id_key] = '_chef_server_session_id'
   c[:session_secret_key]  = Chef::Config.manage_secret_key
Index: chef-0.8.14+dfsg.orig/chef/bin/shef
===================================================================
--- chef-0.8.14+dfsg.orig.orig/chef/bin/shef	2010-05-07 15:06:33.000000000 -0600
+++ chef-0.8.14+dfsg.orig/chef/bin/shef	2010-05-07 15:07:01.000000000 -0600
@@ -19,7 +19,6 @@
 # limitations under the License.
 
 # TODO::EVIL
-require "rubygems"
 require "mixlib/cli"
 
 require "irb"
Index: chef-0.8.14+dfsg.orig/chef-server/config-webui.ru
===================================================================
--- chef-0.8.14+dfsg.orig.orig/chef-server/config-webui.ru	2010-05-07 15:06:33.000000000 -0600
+++ chef-0.8.14+dfsg.orig/chef-server/config-webui.ru	2010-05-07 15:07:01.000000000 -0600
@@ -1,4 +1,3 @@
-require 'rubygems'
 require 'merb-core'
 require 'chef'
 
Index: chef-0.8.14+dfsg.orig/chef-server/config.ru
===================================================================
--- chef-0.8.14+dfsg.orig.orig/chef-server/config.ru	2010-05-07 15:06:33.000000000 -0600
+++ chef-0.8.14+dfsg.orig/chef-server/config.ru	2010-05-07 15:07:01.000000000 -0600
@@ -1,4 +1,3 @@
-require 'rubygems'
 require 'merb-core'
 require 'chef'
 
Index: chef-0.8.14+dfsg.orig/chef-server/bin/chef-server
===================================================================
--- chef-0.8.14+dfsg.orig.orig/chef-server/bin/chef-server	2010-05-07 15:06:33.000000000 -0600
+++ chef-0.8.14+dfsg.orig/chef-server/bin/chef-server	2010-05-07 15:07:01.000000000 -0600
@@ -23,24 +23,14 @@
 # Add chef and chef-server-slice lib dirs to the load path
 # Load chef and chef-server slice from source rather than gem, if present
 
-require "rubygems"
 require "merb-core"
 
-CHEF_SERVER_VERSION = "0.8.14"
+CHEF_SERVER_VERSION = "0.8.14" unless defined?(CHEF_SERVER_VERSION)
 
-[ 'chef', 'chef-server-api' ].each do |lib|
-  $:.unshift(File.expand_path(File.join(File.dirname(__FILE__), "..", "..", "lib")))
-  library = File.expand_path(File.join(File.dirname(__FILE__), "..", "..", lib, "lib", "#{lib}.rb"))
-  require library if File.exists?(library)
-end
+require "chef"
+require "/usr/share/chef-server-api/lib/chef-server-api"
 
-# Ensure the chef gem we load is the same version as the chef server
-unless defined?(Chef)
-  gem "chef", "=" + CHEF_SERVER_VERSION 
-  require 'chef'  
-end
-
-Dir.chdir File.join(File.dirname(__FILE__),"..")
+Dir.chdir File.join(File.dirname(__FILE__),"..", "share", "chef-server")
 __DIR__ = Dir.getwd
 
 if ARGV[0] && ARGV[0] =~ /^[^-]/
Index: chef-0.8.14+dfsg.orig/chef-server/bin/chef-server-webui
===================================================================
--- chef-0.8.14+dfsg.orig.orig/chef-server/bin/chef-server-webui	2010-05-07 15:06:33.000000000 -0600
+++ chef-0.8.14+dfsg.orig/chef-server/bin/chef-server-webui	2010-05-07 15:07:01.000000000 -0600
@@ -23,24 +23,14 @@
 # Add chef and chef-server-slice lib dirs to the load path
 # Load chef and chef-server slice from source rather than gem, if present
 
-require "rubygems"
 require "merb-core"
 
-[ 'chef', 'chef-server-webui' ].each do |lib|
-  $:.unshift(File.expand_path(File.join(File.dirname(__FILE__), "..", "..", "lib")))
-  library = File.expand_path(File.join(File.dirname(__FILE__), "..", "..", lib, "lib", "#{lib}.rb"))
-  require library if File.exists?(library)
-end
+CHEF_SERVER_VERSION = "0.8.14" unless defined?(CHEF_SERVER_VERSION)
 
-CHEF_SERVER_VERSION = "0.8.14"
+require "chef"
+require "/usr/share/chef-server-webui/lib/chef-server-webui"
 
-# Ensure the chef gem we load is the same version as the chef server
-unless defined?(Chef)
-  gem "chef", "=" + CHEF_SERVER_VERSION 
-  require 'chef'  
-end
-
-Dir.chdir File.join(File.dirname(__FILE__),"..")
+Dir.chdir File.join(File.dirname(__FILE__),"..", "share", "chef-server")
 __DIR__ = Dir.getwd
 
 if ARGV[0] && ARGV[0] =~ /^[^-]/
Index: chef-0.8.14+dfsg.orig/chef/bin/chef-client
===================================================================
--- chef-0.8.14+dfsg.orig.orig/chef/bin/chef-client	2010-05-07 15:07:10.000000000 -0600
+++ chef-0.8.14+dfsg.orig/chef/bin/chef-client	2010-05-07 15:07:16.000000000 -0600
@@ -18,7 +18,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-require 'rubygems'
 $:.unshift(File.join(File.dirname(__FILE__), "..", "lib"))
 require 'chef'
 require 'chef/application/client'
Index: chef-0.8.14+dfsg.orig/chef/bin/chef-solo
===================================================================
--- chef-0.8.14+dfsg.orig.orig/chef/bin/chef-solo	2010-05-07 15:07:22.000000000 -0600
+++ chef-0.8.14+dfsg.orig/chef/bin/chef-solo	2010-05-07 15:07:25.000000000 -0600
@@ -18,7 +18,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-require 'rubygems'
 $:.unshift(File.join(File.dirname(__FILE__), "..", "lib"))
 require 'chef/application/solo'
 
Index: chef-0.8.14+dfsg.orig/chef/bin/knife
===================================================================
--- chef-0.8.14+dfsg.orig.orig/chef/bin/knife	2010-05-07 15:07:33.000000000 -0600
+++ chef-0.8.14+dfsg.orig/chef/bin/knife	2010-05-07 15:07:36.000000000 -0600
@@ -18,7 +18,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-require 'rubygems'
 $:.unshift(File.expand_path(File.join(File.dirname(__FILE__), "..", "lib")))
 require 'chef/application/knife'
 
